freva-rest/src/databrowser_api/endpoints.py

```diff
@@ -235,3 +243,78 @@ async def load_data(
         status_code=status_code,
         media_type="text/plain",
     )
+
+
+@app.put(
+    "/api/databrowser/userdata/{username}",
+    status_code=status.HTTP_202_ACCEPTED,
+    tags=["User Data"],
+)
+async def put_user_data(
+    username: str,
+    paths: Annotated[
+        Sequence[str], Query(description="Paths to the user's data to be added")
+    ],
+    facets: Annotated[
+        Mapping[str, str], Body(..., description="Facet key-value pairs for metadata")
+    ],
+    background_tasks: BackgroundTasks,
+    current_user: TokenPayload = Depends(auth.required),
+) -> Dict[str, str]:
+    """Add user data to the Apache Solr search system and MongoDB."""
+    if username != current_user.preferred_username:  # type: ignore
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="Permission denied"
+        )
+
+    solr_instance = Solr(server_config)
+
+    try:
+        validated_paths = await solr_instance._validating_userdata_paths(*paths)
+        background_tasks.add_task(
+            solr_instance.add_userdata, (username, validated_paths, facets)
+        )
+    except FileNotFoundError as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Invalid paths provided: {str(e)}",
+        )
+    except Exception as e:  # pragma: no cover
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to add user data: {str(e)}",
+        )
+    return {
+        "status": "Crawling has been started. Check the Data Browser for updates."
+    }
+
+
+@app.delete(
+    "/api/databrowser/userdata/{username}",
+    status_code=status.HTTP_202_ACCEPTED,
+    tags=["User Data"],
+)
+async def delete_user_data(
+    username: str,
```

Comment:
Same as above about the token, do you need a username if you can take it from the token?